### PR TITLE
net-p2p/soulseek-qt: bump to latest version 2018-1-30

### PR DESCRIPTION
**Pull request by internethering**
- 2016-1-17 sources are offline, removed net-p2p/soulseek-qt-20160117.ebuild
- improved ebuild: download icon and create desktop file
- added new dependency sys-libs/libselinux

Signed-off-by: Richard hering richard@internethering.de
Package-Manager: Portage-2.3.78, Repoman-2.3.17

### CODE REVIEW

File: net-p2p/soulseek-qt/soulseek-qt-20180130.ebuild [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
+inherit desktop
 
 DESCRIPTION="Official binary Qt SoulSeek client"
 HOMEPAGE="http://www.soulseekqt.net/"
 BINARY_NAME="SoulseekQt-${PV:0:4}-$((${PV:4:2}))-$((${PV:6:2}))"
-SRC_URI="
-	x86? ( https://www.dropbox.com/s/kebk1b5ib1m3xxw/${BINARY_NAME}-32bit.tgz )
-	amd64? ( https://www.dropbox.com/s/7qh902qv2sxyp6p/${BINARY_NAME}-64bit.tgz )
-"
+SRC_URI="https://www.slsknet.org/SoulseekQt/Linux/${BINARY_NAME}-64bit-appimage.tgz https://www.dropbox.com/s/3qdatnoi10e3c17/slsk.png"
```
Comment (zx2c4) [inline net-p2p/soulseek-qt/soulseek-qt-20180130.ebuild]: What's this dropbox link?

**Comment by zx2c4** [comment]
Is it possible to extract things from the appimage and use system libraries, or will that be insane?

File: net-p2p/soulseek-qt/soulseek-qt-20180130.ebuild (hunk 1 above)
Comment (internethering) [reply net-p2p/soulseek-qt/soulseek-qt-20180130.ebuild]: this is the slks.png icon, I got this link from: https://www.slsknet.org/news/node/3342

**Comment by internethering** [comment]
> Is it possible to extract things from the appimage and use system libraries, or will that be insane?

`file /opt/bin/soulseek-qt ` says:

`/opt/bin/soulseek-qt: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.18, BuildID[sha1]=d4b0eeecada37bbc753023885a3f0f7e3bdac6cc, stripped`

I don't know how to extract things from that.

**Comment by internethering** [comment]
see: https://bugs.gentoo.org/694548 – you should merge this bug now!

File: net-p2p/soulseek-qt/soulseek-qt-20180130.ebuild [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
 
 LICENSE="free-noncomm"
```
Comment (ulm) [inline net-p2p/soulseek-qt/soulseek-qt-20180130.ebuild]: This is wrong, see [bug 702334](https://bugs.gentoo.org/702334#c3).

Before we can add this, we at least need some statement from upstream about the bundled GPL libraries. Why do they think that this isn't a GPL violation?

**Comment by rpimonitrbtch** [comment]
> 
> 
> > Is it possible to extract things from the appimage and use system libraries, or will that be insane?
> ... 
> I don't know how to extract things from that?

apparently appimages are self-extractable.

./SoulseekQt-2018-1-30-64bit.AppImage --appimage-extract

so something like:

src_unpack() {
        unpack "${BINARY_NAME}-64bit-appimage.tgz"
        "${WORKDIR}/${BINARY_NAME}-64bit.AppImage" --appimage-extract
}

The extracted contents include an icon that could be leveraged, rather than separately fetching the icon from dropbox

**Comment by juippis** [comment]
739b454de428801c64a90f07e2407d3e1222e421 this package has been removed from Gentoo. You can take a look at Gentoo's GURU project in hopes of re-adding it again and becoming the maintainer for it.

**Pull request closed without merging**